pkg/ibm/ibmp.go

```diff
@@ -140,7 +108,60 @@ func (r IBMPowerDynamicConfig) GetInstanceAddress(kubeClient client.Client, ctx
 }
 
 func (r IBMPowerDynamicConfig) ListInstances(kubeClient client.Client, ctx context.Context, instanceTag string) ([]cloud.CloudVMInstance, error) {
-	return nil, fmt.Errorf("not impelemented")
+	log := logr.FromContextOrDiscard(ctx)
+	service, err := r.authenticate(kubeClient, ctx)
+	if err != nil {
+		return nil, err
+	}
+	builder := core.NewRequestBuilder(core.GET).WithContext(ctx)
```

Comment:
can we move the request building mechanics to different functions or even different abstraction objects so that the code here clearly show the logic of what we're actually doing?